q2_longitudinal/tests/test_longitudinal.py

```diff
@@ -915,6 +915,30 @@ def test_maturity_index(self):
             check_series_type=False, check_names=False)
 
 
+class AnovaTests(TestPluginBase):
+    package = 'q2_longitudinal.tests'
+
+    def setUp(self):
+        super().setUp()
+        self.anova = self.plugin.actions['anova']
+        self.md = qiime2.Metadata(
+            pd.DataFrame([[1, 'a', 3], [1, 'b', 7], [1, 'b', 4], [1, 'a', 4],
+                          [1, 'a', 3], [1, 'b', 6], [2, 'b', 14], [2, 'b', 12],
+                          [2, 'b', 13], [2, 'a', 9], [2, 'a', 8], [2, 'a', 9]],
+                         columns=['number', 'letter', 'result'],
+                         index=pd.Index(
+                            ['s1', 's2', 's3', 's4', 's5', 's6', 's7', 's8',
+                             's9', 's10', 's11', 's12'], name='id')))
+
+    def test_execution(self):
+        # does it run?
+        self.anova(self.md, 'result ~ letter+number')
+
+    def test_invalid_formula(self):
```

Comment:
FWIW, the formula parser explodes when the tilde is missing --- this might be a good initial check to perform:

```bash
qiime longitudinal anova --m-metadata-file ~/data/moving-pictures/sample-metadata.tsv --p-formula 'DaysSinceExperimentStart' --o-visualization ~/Desktop/foo.qzv --verbose
Traceback (most recent call last):
  File "/Users/matthew/src/qiime2/q2cli/q2cli/commands.py", line 274, in __call__
    results = action(**arguments)
  File "<decorator-gen-425>", line 2, in anova
  File "/Users/matthew/.conda/envs/q2dev/lib/python3.6/site-packages/qiime2/sdk/action.py", line 231, in bound_callable
    output_types, provenance)
  File "/Users/matthew/.conda/envs/q2dev/lib/python3.6/site-packages/qiime2/sdk/action.py", line 427, in _callable_executor_
    ret_val = self._callable(output_dir=temp_dir, **view_args)
  File "/Users/matthew/src/qiime2/q2-longitudinal/q2_longitudinal/_longitudinal.py", line 246, in anova
    metric, group_columns = _parse_formula(formula)
  File "/Users/matthew/src/qiime2/q2-longitudinal/q2_longitudinal/_utilities.py", line 854, in _parse_formula
    metric = model_desc.lhs_termlist[0].name()
IndexError: list index out of range

Plugin error from longitudinal:

  list index out of range

See above for debug info.
```